packages/db/src/lib/duplicate-detection.ts

```diff
@@ -0,0 +1,92 @@
+import { Knex } from 'knex';
+import { deconstructCollectionId } from '@cumulus/message/Collections';
+import { CollectionPgModel } from '../models/collection';
+
+type GranuleInput = {
+  collectionId: string;
+  producerGranuleId: string;
+  collectionCumulusId?: number;
+  granuleId?: string;
+};
+
+type GranuleRecord = {
+  cumulus_id: number;
+  granule_id: string;
+  producer_granule_id: string;
+  collection_cumulus_id: number;
+  status: string;
+  group_state: string | null;
+};
+
+type DuplicateGranulesResult = {
+  sameCollectionMatches: GranuleRecord[];
+  differentCollectionMatches: GranuleRecord[];
+  customCriteriaMatches: GranuleRecord[];
+};
+
+export const getNextGranuleGroupId = async (knex: Knex) : Promise<number> => {
+  const result = await knex.raw("SELECT nextval('granule_group_id_seq')::int");
+  return result.rows[0].nextval;
+};
+
+/**
+ * Detects "active" duplicate granules based on producerGranuleId and collection.
```

Comment:
Docstring should to explain what the intended behaviors are for the values in GranuleInput, particularly the optional ones.

Reply:
I think we should be really careful with this function name - this isn't returning duplicate granules given a producerGranuleId, it's returning *active* duplicate granules.   Granules in an inactive state are still duplicate granule objects. 

Can we rename this method to reflect that intent?

Reply:
Added docstring and rename function and types [c6638bb](https://github.com/nasa/cumulus/pull/4077/commits/c6638bbcdd59ae192b78df8c5df75177693e9ec3)

Reply:
Looks great, thanks for addressing the comment!